include/kframework/foundry.md

```diff
@@ -532,46 +532,42 @@ Ignore all cheat code calls which take place while `expectRevert` is active.
 ```{.k .bytes}
     rule [foundry.call.ignoreCalls]:
          <k> #call_foundry _ _ => . ... </k>
-         <expected>
-           <expectedRevert> true </expectedRevert>
+         <expectedRevert>
+           <isRevertExpected> true </isRevertExpected>
            ...
-         </expected>
+         </expectedRevert>
       [priority(35)]
 ```
 
-Catch reverts.
-If the current call depth is equal with the `expectedDepth` and the `expectedBytes` match the `<output>` cell, then replace the `EVMC_REVERT` status code with `EVMC_SUCCESS`.
-
 ```{.k .bytes}
-    rule <statusCode> EVMC_REVERT => EVMC_SUCCESS </statusCode>
-         <k> #halt ~> #return _RETSTART _RETWIDTH ... </k>
-         <output> OUT </output>
+    rule [foundry.handleExpectRevert]:
+         <k> (. => #checkRevertReason ~> #clearExpectRevert) ~> #halt ... </k>
+         <statusCode> SC </statusCode>
          <callDepth> CD </callDepth>
-         <expected>
-           <expectedRevert> true => false </expectedRevert>
-           <expectedDepth> CD </expectedDepth>
-           <expectedBytes> EXPECTED </expectedBytes>
-         </expected>
-      requires (EXPECTED =/=K .ByteArray andBool EXPECTED ==K #range(OUT, 4, #sizeByteArray(OUT) -Int 4))
-        orBool EXPECTED ==K .ByteArray
+         <expectedRevert>
+           <isRevertExpected> true </isRevertExpected>
+           <expectedDepth> ED </expectedDepth> 
+           ...
+         </expectedRevert>
+      requires CD <=Int ED
+       andBool SC =/=K EVMC_SUCCESS
```

Comment:
Sholud it be `SC =/=K EVMC_SUCCESS` or `SC ==K EVMC_REVERT`? If it ends with out of gas instead, for example, should we count that as a revert?

Reply:
The only instance of the `Revert` status could that I could find in their rust implementation is here https://github.com/foundry-rs/foundry/blob/master/evm/src/executor/inspector/cheatcodes/mod.rs#L489-L505. I also looked at their `handle_expect_revert` function, but I didn't see any match for the actual status code. But my Rust knowledge is limited.